src/reaction_structure.jl

```diff
@@ -0,0 +1,545 @@
+        prodstoich′ = (T == stoich_type) ? prodstoich : convert.(stoich_type, prodstoich)
+    end
+
+    if !(all(isvalidreactant, subs) && all(isvalidreactant, prods))
+        badsts = union(filter(!isvalidreactant, subs), filter(!isvalidreactant, prods))
+        throw(ArgumentError("""To be a valid substrate or product, non-constant species must be declared via @species, while constant species must be parameters with the isconstantspecies metadata. The following reactants do not follow this convention:\n $badsts"""))
+    end
+
+    ns = if netstoich === nothing
+        get_netstoich(subs, prods, substoich′, prodstoich′)
+    else
+        (netstoich_stoichtype(netstoich) != stoich_type) ?
+        convert.(stoich_type, netstoich) : netstoich
+    end
+
+    # Check that all metadata entries are unique. (cannot use `in` since some entries may be symbolics).
+    if !allunique(entry[1] for entry in metadata)
+        error("Repeated entries for the same metadata encountered in the following metadata set: $([entry[1] for entry in metadata]).")
+    end
+
+    # Deletes potential `:only_use_rate => ` entries from the metadata.
+    if any(:only_use_rate == entry[1] for entry in metadata) 
+        deleteat!(metadata, findfirst(:only_use_rate == entry[1] for entry in metadata))
+    end
+
+    # Ensures metadata have the correct type.
+    metadata = convert(Vector{Pair{Symbol, Any}}, metadata)
+
+    Reaction(value(rate), subs, prods, substoich′, prodstoich′, ns, only_use_rate, metadata)
+end
+
+# Three argument constructor assumes stoichiometric coefs are one and integers.
+function Reaction(rate, subs, prods; kwargs...)
+    sstoich = isnothing(subs) ? nothing : ones(Int, length(subs))
+    pstoich = isnothing(prods) ? nothing : ones(Int, length(prods))
+    Reaction(rate, subs, prods, sstoich, pstoich; kwargs...)
+end
+
+# Union type for `Reaction`s and `Equation`s.
+const CatalystEqType = Union{Reaction, Equation}
+
+### Base Functions ###
+
+# Show function for `Reaction`s.
+function Base.show(io::IO, rx::Reaction)
+    print(io, rx.rate, ", ")
+    print_rxside(io, rx.substrates, rx.substoich)
+    arrow = rx.only_use_rate ? "⇒" : "-->"
+    print(io, " ", arrow, " ")
+    print_rxside(io, rx.products, rx.prodstoich)
+end
+
+function print_rxside(io::IO, specs, stoich)
+    # reactants/substrates
+    if isempty(specs)
+        print(io, "∅")
+    else
+        for (i, spec) in enumerate(specs)
+            prspec = (MT.isparameter(spec) || (MT.operation(spec) == getindex)) ?
+                     spec : MT.operation(spec)
+            if isequal(stoich[i], one(stoich[i]))
+                print(io, prspec)
+            elseif istree(stoich[i])
+                print(io, "(", stoich[i], ")*", prspec)
+            else
+                print(io, stoich[i], "*", prspec)
+            end
+
+            (i < length(specs)) && print(io, " + ")
+        end
+    end
+    nothing
+end
+
+"""
```

Comment:
If you want a comment "title" for the following functions (and I'd move `show` here too) you could call this 

```julia
# dispatches for Base functions.
```

Reply:
That is a good one, will use that